extensions/arc/src/ui/dialogs/connectControllerDialog.ts

```diff
@@ -108,6 +117,7 @@ export class ConnectToControllerDialog extends InitializingComponent {
 		}
 		const controllerInfo: ControllerInfo = {
 			url: url,
+			name: this.nameInputBox.value!,
```

Comment:
The ! shouldn't be necessary I would think

Reply:
It is needed because nameInputBox.value is optional and it can be undefined which we will update later after refreshing controllerModel, I will add a comment.

Reply:
Ah, we're checking username.value above which is why that doesn't need it too.

I'd suggest instead defaulting to an empty string and then changing the line below to 

`controllerModel.info.name = controllerModel.info.name || controllerModel.controllerRegistration?.instanceName;`

Right now if the user had an empty string (they entered text and then deleted it should do that I believe) then the name would be left as the empty string because you're only setting it to the instanceName for undefined/null.

Reply:
ok will do.

Reply:
Fixed. Thanks for the feedback.